pkg/applicationserver/applicationserver.go

```diff
@@ -89,9 +93,25 @@ func New(c *component.Component, conf *Config) (as *ApplicationServer, err error
 	if err != nil {
 		return nil, err
 	}
+
+	ctx := log.NewContextWithField(c.Context(), "namespace", "applicationserver")
+
+	var interopCl InteropClient
+	if conf.Interop.ConfigURI != "" {
+		tlsConf, err := c.GetTLSConfig(ctx)
+		if err == nil {
+			log.FromContext(ctx).WithError(err).Warn("Could not get fallback TLS config for interoperability")
```

Comment:
With `err == nil` and no error handling?

Reply:
It's fallback certificate, which is optional.

Reply:
- You're checking `err == nil` while you use `WithError(err)` and print a warning. So that should be `err != nil`
- If `err != nil`, `tlsConf` is undefined. I don't care that the current implementation of `TLSConfig` happens to return `nil` on error